pkg/fastkmeans.c

```
@@ -0,0 +1,70 @@
+        ++res;
+    }
+    return res - 1;
+}
+
+void kmeansIters(int64_t clustersCenters[][3], size_t clustersCentersLen, int64_t pixelColors[][3], size_t pixelColorsLen) {
+	size_t batchMaxSize = ssqrt(pixelColorsLen);
+    size_t sumAndCountLen = sizeof(int64_t) * clustersCentersLen * 4;
+	int64_t* sumAndCount = malloc(sumAndCountLen); // count and sum of Rs, Gs, Bs
+	for (int epoch = 0; epoch < 300; ++epoch) {
+		size_t k = (size_t)rand() % (batchMaxSize + 1);
+        memset(sumAndCount, 0, sumAndCountLen);
```

Cppcheck (2.19.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)